packages/common/core-interfaces/src/error.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryProperties } from "./index";
+
+/**
+ * Different error types the Container may report out to the Host
+ */
+export const FluidErrorTypes = {
+	/**
+	 * Some error, most likely an exception caught by runtime and propagated to container as critical error
+	 */
+	genericError: "genericError",
+
+	/**
+	 * Throttling error from server. Server is busy and is asking not to reconnect for some time
+	 */
+	throttlingError: "throttlingError",
+
+	/**
+	 * Data loss error detected by Container / DeltaManager. Likely points to storage issue.
+	 */
+	dataCorruptionError: "dataCorruptionError",
+
+	/**
+	 * Error encountered when processing an operation. May correlate with data corruption.
+	 */
+	dataProcessingError: "dataProcessingError",
+
+	/**
+	 * Error indicating an API is being used improperly resulting in an invalid operation.
+	 */
+	usageError: "usageError",
+} as const;
+export type FluidErrorTypes = typeof FluidErrorTypes[keyof typeof FluidErrorTypes];
+
+/**
+ * Base interface for all errors and warnings at container level
+ */
+export interface IErrorBase extends Partial<Error> {
+	/** errorType is a union of error types from
+	 * - container
+	 * - runtime
+	 * - drivers
+	 */
+	readonly errorType: string;
+
+	/**
+	 * See Error.message
+	 * Privacy Note - This is a freeform string that we may not control in all cases (e.g. a dependency throws an error)
+	 * If there are known cases where this contains privacy-sensitive data it will be tagged and included in the result
+	 * of getTelemetryProperties. When logging, consider fetching it that way rather than straight from this field.
+	 */
+	readonly message: string;
+	/** See Error.name */
+	readonly name?: string;
+	/** See Error.stack */
+	readonly stack?: string;
+	/**
+	 * Returns all properties of this error object that are either safe to log
+	 * or explicitly tagged as containing privacy-sensitive data.
```

Comment:
Should this say "not explicitly tagged as containing privacy-sensitive data"? @markfields Maybe you can answer this. This seems wrong to me.

Reply:
Doesn't necessarily need to be resolved in this PR, since we didn't change the contents here. Still probably good to get an answer and update as needed when we can.